### PR TITLE
Add support for conditionals

### PR DESCRIPTION
**Pull request by alexschneider**
Right now, this allows for the following:

```
if true: out("false")
else if false:
  out("true")
else: out("maybe")
```

because support for that is trivial to add (unlike we feared). If anyone doesn't like that, we can easily only only allow blocks for combined if, else if, else expressions, and only allow the single line for if expressions alone.

### CODE REVIEW

**Comment by alexschneider** [comment]
To save headaches in the future, if people install new modules, the npm-shrinkwrap.json needs to be updated!

File: parser/start_tokens.coffee [hunk 1]
```diff
@@ -12,7 +12,7 @@ class StartTokens
   @expression = [
     'INTLIT', 'FLOATLIT', 'true', 'false', # Primitive literals
     'ID', '->', 'trait', 'class',          # Misc other tokens
-    @iterable, tokens.reservedWords,
+    @iterable, tokens.beginnings,
```
Comment (rachelriv) [inline parser/start_tokens.coffee]: I'd rather have beginning only be specified in `start_tokens` and keep the reserved words in the scanner as they were before. I don't think logic for the parser should spill into any part of the scanner.

File: scanner/tokens.coffee [hunk 2]
```diff
@@ -17,21 +17,26 @@ class Tokens
     '+=', '-=', '*=', '/=', '++', '--',  # mutators
     '<=', '>=',                          # relational operators
     '->',                                # function
-    '..',                                # range
+    '..', 'by',                          # range
     ':='                                 # declaration
   ]
 
-  @reservedWords = [
+  @beginnings = [
```
Comment (rachelriv) [inline scanner/tokens.coffee]: See comment above

File: test/parser/parser_spec.coffee [hunk 3]
```diff
@@ -82,4 +82,15 @@ describe 'Parser', ->
...
+  
+  describe 'parsing a valid conditional expression', ->
+    context 'when a conditional has if, else, and else if branches', ->
+
+      it 'parses correctly', (done) ->
+        scan "#{validProgramsPath}/program13.tea", (err, tokens) ->
+          {error, program} = parse tokens
+          expect(error).to.be.null
+          expect(program.toString()).to.eql outputASTs.program13_AST_string
+          done()
```
Comment (rachelriv) [inline test/parser/parser_spec.coffee:88]: Great test adding! :+1:

File: parser/start_tokens.coffee (hunk 1 above)
Comment (alexschneider) [inline parser/start_tokens.coffee]: Do you mind submitting an alternative PR to #50? I'm unclear what you want to do exactly.

Comment (rachelriv) [inline parser/start_tokens.coffee]: Sure thing! I'll have to do it a little later today.
But everything else in this PR looks good to me.

Comment (rachelriv) [inline parser/start_tokens.coffee]: If you merge master into this branch and update the handling of start tokens, this branch should be good to go!

**Comment by alexschneider** [comment]
@rachelriv Updated, please merge

File: scanner/tokens.coffee (hunk 2 above)
Comment (rachelriv) [inline scanner/tokens.coffee]: Can we change this back to `reservedWords` please?

File: scanner/tokens.coffee [hunk 4]
```diff
@@ -17,21 +17,26 @@ class Tokens
...
-    'if', 'else',                        # if/then
+    'for', 'while',                      # loops/blocks/range keywords
+    'if',                                # if/then
     'not',                               # negation
     'true', 'false',                     # boolean literals
     'new',                               # instantiate a class/create an object
     'return'                             # we need to return from functions
   ]
 
+  @otherReservedWords = [
```
Comment (rachelriv) [inline scanner/tokens.coffee]: It looks like this is still the old implementation.

Comment (alexschneider) [inline scanner/tokens.coffee]: Fixed

**Comment by rachelriv** [comment]
@alexschneider LGTM but can you please rebase so it can be merged automatically?

**Comment by alexschneider** [comment]
@rachelriv done

**Comment by rachelriv** [comment]
It looks like one of your parser tests if failing. This should be good once all tests are passing.

**Comment by rachelriv** [comment]
LGTM.

**Pull request merged**